tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/CDIBean.java

```diff
@@ -0,0 +1,170 @@
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ManagedExecutorConfig;
+import org.eclipse.microprofile.concurrent.NamedInstance;
+import org.testng.Assert;
+
+@RequestScoped
+public class CDIBean {
+
+    @Inject
+    ManagedExecutor exec;
+
+    @Inject
+    @NamedInstance("maxAsync2")
+    @ManagedExecutorConfig(maxAsync = 2)
+    ManagedExecutor maxAsync2;
+
+    @Inject
+    @NamedInstance("appProduced")
+    ManagedExecutor appProduced;
+
+    @Inject
+    @NamedInstance("appProduced_injected")
+    ManagedExecutor appProduced_injected;
+
+    @Inject
+    ManagedExecutor throwAway1;
+
+    @Inject
+    ManagedExecutor throwAway2;
+
+    @Produces
+    @ApplicationScoped
+    @NamedInstance("appProduced")
+    public ManagedExecutor createExec() {
+        return ManagedExecutor.builder().build();
+    }
+
+    @Produces
+    @ApplicationScoped
+    @NamedInstance("appProduced_injected")
+    public ManagedExecutor createExecInjected(@ManagedExecutorConfig(maxAsync = 2) ManagedExecutor exec) {
+        System.out.println("@AGG in producer with exec: " + exec);
+        Thread.dumpStack();
+        return exec;
+    }
+
+    /**
+     * Verify that injected ME instances injected by the container can be shutdown
+     */
+    public void shutdownContainerInstance() throws Exception {
+        throwAway1.shutdown();
+        assertTrue(throwAway1.awaitTermination(5, TimeUnit.SECONDS));
+        assertTrue(throwAway1.isShutdown());
+
+        throwAway2.shutdownNow();
+        assertTrue(throwAway2.awaitTermination(5, TimeUnit.SECONDS));
+        assertTrue(throwAway1.isShutdown());
+    }
+
+    /**
+     * Extra sanity check test to verify injection is occurring. However, if CDI is 
+     * set up properly, this bean should not even be reachable if injection fails. 
+     */
+    public void testVerifyInjection() {
+        assertNotNull(exec);
+        assertNotNull(maxAsync2);
+        assertNotNull(appProduced);
+        assertNotNull(appProduced_injected);
+    }
+
+    /**
+     * Verify that injected ME instances are useable in a very basic way
+     */
+    public void testBasicExecutorUsable() throws Exception {
+        assertEquals(exec.supplyAsync(() -> "hello").get(5, TimeUnit.SECONDS), "hello");
+        assertEquals(maxAsync2.supplyAsync(() -> "hello").get(5, TimeUnit.SECONDS), "hello");
+        assertEquals(appProduced.supplyAsync(() -> "hello").get(5, TimeUnit.SECONDS), "hello");
+        assertEquals(appProduced_injected.supplyAsync(() -> "hello").get(5, TimeUnit.SECONDS), "hello");
+    }
+
+    /**
+     * Verify that the @ManagedExecutorConfig annotation is applied to injection points
+     */
+    public void testMaxAsync2() throws Exception {
+        checkMaxAsync2(maxAsync2);
+        checkMaxAsync2(appProduced_injected);
+    }
+    
+    private void checkMaxAsync2(ManagedExecutor exec) throws Exception {
```

Comment:
Verifying maxAsync is less reliable than verifying maxQueued because it involves waiting for a fixed amount of time and observing whether additional tasks might have started.  This is both wasteful of test execution time and not guaranteed to actually catch errors.  I understand why we have one test that does this (which I wrote in a different test class and which this partly seems to copy from) because we want at least some amount of verification.  But I don't see why we would want to repeat this test multiple times as a way of verifying that `@ManagedExecutorConfig` in general was processed.  This should not be a utility method for repeated usage.  It should only be done once.  To verify the general case that `@ManagedExecutorConfig` was applied, it would be better to use maxQueued, propagated, or cleared.

Reply:
good point, I'll use maxQueued instead since that is more efficient